backup.py

```diff
@@ -0,0 +1,142 @@
+
+
+def sendEmail(message=str,subject_tag="success"):
+    email = f"Subject: Backup alerts - {subject_tag}\n\n{message}"
+
+    try:
+        #send your message with credentials specified above
+
+        if EMAIL_USE_TLS:
+            logging.debug("Using SMTP-TLS")
+            server = smtplib.SMTP(EMAIL_HOST,int(EMAIL_PORT))
+            server.starttls()
+            server.ehlo()
+        else:
+            logging.debug("Using SMTP-SSL")
+            server = smtplib.SMTP_SSL(EMAIL_HOST,int(EMAIL_PORT))
+            
+
+        server.login(EMAIL_USER, EMAIL_PASS)
+        server.sendmail(EMAIL_FROM, EMAIL_TO, email)
+        server.close()
+
+        logging.debug("Email sent successfully")
+    except (gaierror, ConnectionRefusedError):
+        logging.error('Error while sending email: the SMTP server refused the connection')
+    except smtplib.SMTPServerDisconnected:
+        logging.error('Error while sending email, check login credentials')
+    except smtplib.SMTPException as e:
+        logging.error('Error whiel sending email:  General SMTP error: ' + str(e))
+
+
+
+
+os.environ["BORG_PASSPHRASE"]=BACKUP_ENCRYPTION_KEY
+os.environ["BORG_HOST_ID"]=BACKUP_NAME
+os.environ["BORG_REPO"]=BORG_REPO
+
+if BACKUP_VERBOSE:
+    logging.basicConfig(level=logging.DEBUG)
+else:
+    logging.basicConfig(level=logging.INFO)
+
+if EMAIL_TEST:
+    logging.info("Sending test email")
+    sendEmail("Test email",subject_tag="TEST")
+    exit(0)
+
+
+logging.debug("Starting Borg Backup")
+
+if not os.path.isdir(BORG_REPO):
+    logging.debug("Starting Repository Initialization")
+    os.mkdir(BORG_REPO)
+
+    command = ['borg', 'init', '--encryption=repokey']
+    logging.debug("Executing command: " + str(command))
+    try:
+        output = subprocess.run(command, shell=False,capture_output=True, check=True)
+    except subprocess.CalledProcessError as error:
+        logging.fatal("There was a problem initializing the repository")
+        sendEmail("Error during repository initialization, borg said: " + error.stderr, subject_tag="ERROR_BORG")
+        raise Exception("Error running command: " + error.stderr)
+    logging.debug("Ending Repository Initialization")
+
+
+borg_pid = subprocess.run(['pidof','borg'],shell=False,capture_output=True)
+if not borg_pid.stdout:
+    logging.debug("Clearing Locks; Borg Not Running")
+    subprocess.run(['borg','break-lock'],shell=False,check=True)
+
+
+logging.debug("Starting Daily Archive")
+timestamp = strftime("%Y-%m-%d-%s", localtime())
+command = ['borg','create','::'+timestamp,'/data']
+if BORG_CUSTOM_ARGS:
+    command[2:0] = BORG_CUSTOM_ARGS
+
+logging.debug("Executing command: " + str(command))
+try:
+    output = subprocess.run(command,shell=False,check=True,capture_output=True)
+except subprocess.CalledProcessError as error:
+    logging.fatal("There was a problem creating the daily archive")
+    sendEmail("Error during archive creation, borg said: " + error.stderr, subject_tag="ERROR_BORG")
```

Comment:
There appears to be an error concatenating stderr. I ran two instances just to see what would happen during an error:
```
Traceback (most recent call last):
  File "/app/./backup.py", line 108, in <module>
    sendEmail("Error during archive creation, borg said: " + error.stderr, subject_tag="ERROR_BORG")
TypeError: can only concatenate str (not "bytes") to str
```

I imagine the other calls to `sendEmail` that use stderr might have it as well?

Reply:
Yep, I always forget to to this for some reason.  Fixed.